dot/parachain/overseer/overseer.go

```diff
@@ -215,24 +216,16 @@ func (o *OverseerSystem) handleBlockEvents() {
 			o.activeLeaves[imported.Header.Hash()] = uint32(imported.Header.Number)
 			delete(o.activeLeaves, imported.Header.ParentHash)
 
-			// TODO:
-			/*
-				- Add active leaf only if given head supports parachain consensus.
-				- You do that by checking the parachain host runtime api version.
-				- If the parachain host runtime api version is at least 1, then the parachain consensus is supported.
-
-					#[async_trait::async_trait]
-					impl<Client> HeadSupportsParachains for Arc<Client>
-					where
-						Client: RuntimeApiSubsystemClient + Sync + Send,
-					{
-						async fn head_supports_parachains(&self, head: &Hash) -> bool {
-							// Check that the `ParachainHost` runtime api is at least with version 1 present on chain.
-							self.api_version_parachain_host(*head).await.ok().flatten().unwrap_or(0) >= 1
-						}
-					}
-
-			*/
+			supports, err := o.headSupportsParachain(imported.Header.Hash())
+			if err != nil {
+				logger.Criticalf("checking head supports parachain: %s", err.Error())
+				return
+			}
+
+			if !supports {
+				return
```

Comment:
might be better to add a log and return. My feeling is that silent returns generally make code confusing to debug.